Firestore/Swift/Source/Codable/DocumentSnapshot+ReadDecodable.swift

```diff
@@ -38,4 +42,28 @@ extension DocumentSnapshot {
     }
     return nil
   }
+
+  /// Retrieves all fields in a document and converts them to an instance of
+  /// caller-specified type. Returns `nil` if the document does not exist.
+  ///
+  /// See `Firestore.Decoder` for more details about the decoding process.
+  ///
+  /// - Parameters
+  ///   - type: The type to convert the document fields to.
+  ///   - serverTimestampBehavior: Configures how server timestamps that have
+  ///     not yet been set to their final value are returned from the snapshot.
+  ///   - decoder: The decoder to use to convert the document. `nil` to use
+  ///     default decoder.
+  public func data<T: Decodable>(as type: T.Type,
+                                 with serverTimestampBehavior: ServerTimestampBehavior,
+                                 decoder: Firestore.Decoder? = nil) throws -> T? {
+    var d = decoder
+    if d == nil {
+      d = Firestore.Decoder()
+    }
```

Comment:
Drive-by nit: You can remove an optional in this code with

```swift
let d = decoder ?? Firestore.Decoder() // can use var instead of let if d must be mutable
```

Reply:
Nice! Done.